tools/windows/install-help/cal/customactiondata.cpp

```diff
@@ -34,24 +34,26 @@ bool CustomActionData::init(const std::wstring &data)
         return false;
     }
 
-    auto start = data.begin();
-    auto end = data.end();
-    std::wregex re(L"((\\w+)=(.+)?;\\s*\r?\n)");
-    std::match_results<decltype(start)> results;
-    while (std::regex_search(start, end, results, re))
+    std::wstringstream ss(data);
+    std::wstring token;
+
+    while (std::getline(ss, token))
     {
-        auto propertyValue = results[3].str();
-        propertyValue.erase(propertyValue.begin(), std::find_if(propertyValue.begin(), propertyValue.end(), [](int ch)
-        {
-            return !std::isspace(ch);
-        }));
-        if (propertyValue.length() > 0)
+        // now 'token'  has the key=val; do the same thing for the key=value
+        bool boolval = false;
+        std::wstringstream instream(token);
+        std::wstring key, val;
+        if (std::getline(instream, key, L'='))
         {
-            values[results[2]] = propertyValue;
+            trim_string(key);
+            std::getline(instream, val);
+            trim_string(val);
+            if (key.length() > 0 && val.length() > 0)
```

Comment:
```suggestion
            if (!key.empty() && !val.empty())
```
I'm not convinced about which expression is more legible. Feel free to ignore this change.